secret/secrets_manager_client_test.go

```diff
@@ -61,11 +72,77 @@ func TestSecretsManagerCreateAndDeleteSecret(t *testing.T) {
 		require.NoError(t, err)
 		require.NotZero(t, out)
 
+		defer Cleanup(out)
+
+		require.NotZero(t, out.Name)
+		require.NotZero(t, out.ARN)
+
+	})
+
+	t.Run("GetFailsWithInvalidInput", func(t *testing.T) {
+		out, err := c.GetSecretValue(ctx, &secretsmanager.GetSecretValueInput{})
+		assert.Error(t, err)
+		assert.Zero(t, out)
+	})
+
+	t.Run("UpdateFailsWithInvalidInput", func(t *testing.T) {
+		out, err := c.UpdateSecret(ctx, &secretsmanager.UpdateSecretInput{})
+		assert.Error(t, err)
+		assert.Zero(t, out)
+	})
+
+	t.Run("CreateAndGetSucceed", func(t *testing.T) {
+		out, err := c.CreateSecret(ctx, &secretsmanager.CreateSecretInput{
+			Name:         aws.String(os.Getenv("AWS_SECRET_PREFIX") + "foobar"),
+			SecretString: aws.String("barfoo"),
+		})
+		require.NoError(t, err)
+		require.NotZero(t, out)
+
+		defer Cleanup(out)
+
+		defer func() {
```

Comment:
I would not do this in a defer - defers are typically for cleanup, not for testing code. It's also problematic because if someone adds code to the test, they expect it to run top to bottom. The defer is implicitly assuming that the secret is still valid, which it might not be (e.g. it might be modified).